### PR TITLE
[develop < T0122] add AllShortest support

### PR DESCRIPTION
**Pull request by brunos252**
### Description

Added support for the new all shortest paths algorithm to integrated_algorithms.py

### Pull request type

- [x] Feature

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

######################################

### Reviewer checklist (the reviewer checks this part)
- [ ] Core feature implementation
- [ ] Tests
- [ ] Code documentation
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)

######################################

### CODE REVIEW

**Comment by Josipmrden** [comment]
Did you run and commit changes from `poetry install` perhaps?

**Comment by BorisTasevski** [comment]
Hold on until [this one is resolved](https://github.com/memgraph/gqlalchemy/pull/182). It will solve Your build

**Review by BorisTasevski — APPROVED** [review]

File: gqlalchemy/graph_algorithms/integrated_algorithms.py [hunk 1]
```diff
@@ -201,3 +202,54 @@ def __str__(self) -> str:
...
+    ) -> None:
+        """
+        Args:
+            upper_bound: Upper bound for path depth. Defaults to None.
+            condition: Filter through nodes and relationships that pass this
+            condition. Defaults to None.
+            total_weight_var: Variable defined as the sum of all weights on
+            path being returned. Defaults to "total_weight".
+            weight_property: property being used as weight. Defaults to
+            "r.weight".
```
Comment (katarinasupe) [inline gqlalchemy/graph_algorithms/integrated_algorithms.py:236]: We didn't write to what it defaults so far. Follow the guidelines like in other PR. Also, start all arguments description with an uppercase.

File: gqlalchemy/graph_algorithms/integrated_algorithms.py [hunk 2]
```diff
@@ -201,3 +202,54 @@ def __str__(self) -> str:
...
+        """
+        super().__init__()
+        self.weight_property = f"r.{weight_property}" if "." not in weight_property else weight_property
```
Comment (katarinasupe) [inline gqlalchemy/graph_algorithms/integrated_algorithms.py:239]:
```suggestion
        self.weight_property = weight_property if "." in weight_property else f"r.{weight_property}"
```
I think we talked about negations in other PRs, the code is more readable if you avoid negation here.

File: gqlalchemy/graph_algorithms/integrated_algorithms.py [hunk 3]
```diff
@@ -201,3 +202,54 @@ def __str__(self) -> str:
...
+        self.total_weight_var = total_weight_var
+        self.condition = condition
+        self.upper_bound = str(upper_bound) if upper_bound is not None else ""
```
Comment (katarinasupe) [inline gqlalchemy/graph_algorithms/integrated_algorithms.py:242]:
```suggestion
        self.upper_bound = "" if upper_bound is None else str(upper_bound)
```

Again the same. Correct my code if I am wrong, writing in comments.

File: gqlalchemy/graph_algorithms/integrated_algorithms.py [hunk 4]
```diff
@@ -201,3 +202,54 @@ def __str__(self) -> str:
...
+
+    def __str__(self) -> str:
+        algo_str = ALLSHORTEST_EXPANSION
+        if self.upper_bound != "":
```
Comment (katarinasupe) [inline gqlalchemy/graph_algorithms/integrated_algorithms.py:246]: Not sure, but did we leave an empty line above `if`s? If yes (pep8), please fix it everywhere.

**Review by katarinasupe — CHANGES_REQUESTED** [review]

File: gqlalchemy/graph_algorithms/integrated_algorithms.py (hunk 4 above)
Comment (brunos252) [reply gqlalchemy/graph_algorithms/integrated_algorithms.py:246]: https://peps.python.org/pep-0008/#blank-lines
they may be used, for readability to divide code blocks

File: gqlalchemy/graph_algorithms/integrated_algorithms.py [hunk 5]
```diff
@@ -158,6 +158,7 @@ def to_cypher_bounds(self) -> str:
 
 class WeightedShortestPath(IntegratedAlgorithm):
     """Build a Djikstra shortest path call for a Cypher query
```
Comment (katarinasupe) [inline gqlalchemy/graph_algorithms/integrated_algorithms.py:160]: Just add a full stop at the end of the sentence. I know I am boring ❤️

**Review by katarinasupe — CHANGES_REQUESTED** [review]

**Comment by Josipmrden** [comment]
Is there some kind of blocker on github actions?

**Comment by BorisTasevski** [comment]
> Is there some kind of blocker on github actions?

Should simply merge develop into the branch

**Review by katarinasupe — APPROVED** [review]

**Comment by vpavicic** [comment]
@brunos252: All shortest paths algorithm can now be run by.... 

I would just add how to run this algorithm (if it's a short command)

Or maybe

Besides BSF, DSF and WSHORTEST, you can now also run All shortest paths algorithm.

**Comment by brunos252** [comment]
First we make an allShortest object with parameters, and then pass it to QueryBuilder. So it could be a one liner but better not, and it wouldn't be that short if we want 1-2 sentence release note.
So your second proposition seems good to me

**Pull request merged**